packages/jsii-spec/lib/spec.ts

```diff
@@ -484,13 +493,17 @@ export interface Parameter extends Documentable {
     name: string;
 
     /**
-     * The type of the parameter.
+     * The specification for the value of this parameter (includes it's declared type and whether it is optional).
      */
-    type: TypeReference;
+    value: TypeInstance<TypeReference>;
```

Comment:
A parameter can never be a `Promise<T>`. It *can* be optional, however.

Reply:
```ts
// In fact, totally possible:
async function impossible(arg: Promise<Foo>): Promise<Bar>;
```

Reply:
Sure. What I meant to say is that the kernel does not have adequate provisions to handle this, so we shouldn't allow it either.